src/packagedcode/phpcomposer.py

```diff
@@ -0,0 +1,403 @@
+    'git://',
+    'git+git://',
+    'hg+https://',
+    'hg+http://',
+    'git+https://',
+    'git+http://',
+    'svn+https://',
+    'svn+http://',
+    'svn://',
+)
+
+
+def parse_repo_url(repo_url):
+    """
+    Validate a repo_ulr and handle shortcuts for GitHub, GitHub gist,
+    Bitbucket, or GitLab repositories:
+
+    See https://getcomposer.org/doc/04-schema.md#repositories
+
+    These should be resolved:
+        gist:11081aaa281
+        bitbucket:example/repo
+        gitlab:another/repo
+        expressjs/serve-static
+        git://github.com/angular/di.js.git
+        git://github.com/hapijs/boom
+        git@github.com:balderdashy/waterline-criteria.git
+        http://github.com/ariya/esprima.git
+        http://github.com/isaacs/nopt
+        https://github.com/chaijs/chai
+        https://github.com/christkv/kerberos.git
+        https://gitlab.com/foo/private.git
+        git@gitlab.com:foo/private.git
+    """
+
+    is_vcs_url = repo_url.startswith(VCS_URLS)
+    if is_vcs_url:
+        # TODO: ensure the .git suffix is present if needed
+        return repo_url
+
+    if repo_url.startswith('git@'):
+        left, right = repo_url.split('@', 1)
+        host, repo = right.split(':', 1)
+        # may be we should
+        if any(h in host for h in ['github', 'bitbucket', 'gitlab']):
+            return 'https://%(host)s/%(repo)s' % locals()
+        else:
+            return repo_url
+
+    if repo_url.startswith('gist:'):
+        return repo_url
+
+    elif repo_url.startswith(('bitbucket:', 'gitlab:', 'github:')):
+        hoster_urls = {
+            'bitbucket:': 'https://bitbucket.org/%(repo)s',
+            'github:': 'https://github.com/%(repo)s',
+            'gitlab:': 'https://gitlab.com/%(repo)s',
+        }
+        hoster, repo = repo_url.split(':', 1)
+        return hoster_urls[hoster] % locals()
+    elif len(repo_url.split('/')) == 2:
+        # implicit github
+        return 'https://github.com/%(repo_url)s' % locals()
+    return repo_url
+
+
+def deps_mapper(deps, package, field_name):
+    """
+    Handle deps such as dependencies, devDependencies
+    return a tuple of (dep type, list of deps)
+    https://getcomposer.org/doc/04-schema.md#package-links
+    """
+    dep_types = {
+        'dependencies': models.dep_runtime,
+        'devDependencies': models.dep_dev,
+    }
+    resolved_type = dep_types[field_name]
+    dependencies = []
+    for name, version_constraint in deps.items():
+        dep = models.Dependency(name=name, version_constraint=version_constraint)
+        dependencies.append(dep)
+    if resolved_type in package.dependencies:
+        package.dependencies[resolved_type].extend(dependencies)
+    else:
+        package.dependencies[resolved_type] = dependencies
+    return package
+
+
+dependencies_mapper = partial(deps_mapper, field_name='dependencies')
+dev_dependencies_mapper = partial(deps_mapper, field_name='devDependencies')
+
+
+person_parser = re.compile(
```

Comment:
I reckon this was needed for the weirdness of NPMs, but why would this apply to composer?